akka-persistence-typed/src/test/scala/akka/persistence/typed/ManyRecoveriesSpec.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed
+
+import akka.actor.typed.TypedAkkaSpecWithShutdown
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.scaladsl.adapter.TypedActorSystemOps
+import akka.persistence.typed.scaladsl.PersistentBehaviors.CommandHandler
+import akka.persistence.typed.scaladsl.{ Effect, PersistentBehavior, PersistentBehaviors }
+import akka.testkit.TestLatch
+import akka.testkit.typed.scaladsl.{ ActorTestKit, TestProbe }
+import com.typesafe.config.{ Config, ConfigFactory }
+import org.scalatest.concurrent.Eventually
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+object ManyRecoveriesSpec {
+
+  sealed case class Cmd(s: String)
+
+  final case class Evt(s: String)
+
+  def persistentBehavior(
+    name:  String,
+    probe: TestProbe[String],
+    latch: Option[TestLatch]): PersistentBehavior[Cmd, Evt, String] =
+    PersistentBehaviors.receive[Cmd, Evt, String](
+      persistenceId = name,
+      initialState = "",
+      commandHandler = CommandHandler.command {
+        case Cmd(s) ⇒ Effect.persist(Evt(s)).andThen(_ ⇒ probe.ref ! s"$name-$s")
+      },
+      eventHandler = {
+        case (state, _) ⇒ latch.foreach(Await.ready(_, 10.seconds)); state
+      }
+    )
+
+  def forwardBehavior(sender: TestProbe[String]): Behaviors.Receive[Int] =
+    Behaviors.receivePartial[Int] {
```

Comment:
`receiveMessagePartial` since you don't care about the context here
(yeah, it's new)

Reply:
Fixed